lib/computer_receiver.rb

```diff
@@ -1,19 +1,15 @@
 class ComputerReceiver
-  def initialize(validator, converter)
-    @display = display
-    @validator = validator
-
+  def initialize(converter)
     @converter = converter
   end
 
   def input
     rand(1..9)
   end
 
-  def get_player_move(board, mark)
+  def get_player_input(validator, display)
     move = input
-
-    until @validator.valid_move?(board, move, @converter)
+    until validator.valid?(move)
       move = input
     end
     @converter.numberfy(move)
```

Comment:
Non-blocking but you can also initialize this validator with the converter just like you do with the move validator.